src/scf/self_consistent_field.jl

```diff
@@ -62,6 +66,27 @@ function scf_convergence_density_difference(tolerance)
     info -> norm(info.ρout.fourier - info.ρin.fourier) < tolerance
 end
 
+"""
+Determine the tolerance used for the next diagonalisation. This function takes
+``|ρnext - ρin|`` and multiplies it with `ratio_ρdiff` to get the next `diagtol`,
+ensuring additionally that the returned value is between `diagtol_min` and `diagtol_max`.
+"""
+function scf_determine_diagtol(ratio_ρdiff=0.2, diagtol_min=nothing, diagtol_max=0.1)
```

Comment:
call it `default_determine_diagtol`?

Reply:
Nah ... I like `scf_determine_diagtol` more. But this will probably be renamed once we do the SCF params properly.

Reply:
The `default` has the merit of being symmetrical with the SCF convergence criteria. What i don't like with what we have now is that it's unclear that these are functions returning function. We should perphaps capitalize them / make them structs

Reply:
Structs I don't like for such simple stuff. Capitalisation is ok.

Reply:
It's more of an outside API thing. The convention in packages seems to crystallize around options/methods/controllers being structs (eg optim, interpolations, etc.), which are capitalized